samples/buildstrategy/buildpacks-v3/buildstrategy_buildpacks-v3-heroku_cr.yaml

```
@@ -44,12 +44,27 @@ spec:
 
           envs=($(env))
 
+          # Denying the creation of non required files from system environments.
+          # The creation of a file named PATH (corresponding to PATH system environment)
+          # caused failure for python source with requirements.txt (https://github.com/Azure-Samples/python-docs-hello-world)
+          block_list=("PATH" "HOSTNAME" "PWD" "_" "SHLVL" "HOME")
+
           for env in "${envs[@]}"; do
-              IFS='=' read -r key value string <<< "$env"
-              if [[ "$key" != "" ]]; then
-                  path="${ENV_DIR}/${key}"
-                  echo -n "$value" > "$path"
-              fi
+            flag=false
+
+            IFS='=' read -r key value string <<< "$env"
+
+            for str in "${block_list[@]}"; do
+                if [[ "$key" == "$str" ]]; then
+                    flag=true
+                    break
+                fi
```

Indendation is inconsistent. You had previously two, here now four characters.
```suggestion
              if [[ "$key" == "$str" ]]; then
                blocked=true
                break
              fi
```